p2p/proxy.go

```diff
@@ -0,0 +1,92 @@
+package p2p
```

Comment:
As this doesn't really depend on any of the p2p internals, I'd move this to `p2p/proxy`

Reply:
Ah something went awry here; this should have been  in ``core/corehttp``, now moved.